pkg/agent/controller/noderoute/node_route_controller.go

```diff
@@ -441,9 +449,9 @@ func (c *Controller) addNodeRoute(nodeName string, node *corev1.Node) error {
 		// stale routes, flows, and relevant cache of this podCIDR are removed appropriately, we wait for the Node deletion
 		// event to be processed before proceeding, or the route installation and uninstallation operations may override or
 		// conflict with each other.
-		if len(nodesHaveSamePodCIDR) > 0 {
+		if len(nodesHaveSamePodCIDR) > 0 && (len(nodesHaveSamePodCIDR) != 1 || nodesHaveSamePodCIDR[0] != nodeName) {
```

Comment:
Yes. Updated.

Reply:
Maybe "For Windows Noencap case, it is possible that nodesHaveSamePodCIDR is the Node itself because the Node may be processed more than once if its MAC annotation is updated."

Reply:
Updated.